sdk/lib/opentelemetry/sdk/trace/tracer_provider.rb

```diff
@@ -64,14 +63,24 @@ def tracer(name = nil, version = nil)
         # After this is called all the newly created {Span}s will be no-op.
         #
         # @param [optional Numeric] timeout An optional timeout in seconds.
+        # @return [Integer] Export::SUCCESS if no error occurred, Export::FAILURE if
+        #   a non-specific failure occurred, Export::TIMEOUT if a timeout occurred.
         def shutdown(timeout: nil)
           @mutex.synchronize do
             if @stopped
               OpenTelemetry.logger.warn('calling Tracer#shutdown multiple times.')
-              return
+              return Export::FAILURE
+            end
+
+            start_time = OpenTelemetry::Common::Utilities.timeout_timestamp
+            results = @span_processors.map do |processor|
+              remaining_timeout = OpenTelemetry::Common::Utilities.maybe_timeout(timeout, start_time)
+              break [Export::TIMEOUT] if remaining_timeout&.zero?
+
+              processor.shutdown(timeout: remaining_timeout)
             end
-            @active_span_processor.shutdown(timeout: timeout)
             @stopped = true
+            results.uniq.max || Export::SUCCESS
```

Comment:
Do you need to `uniq` before `max`?

Reply:
No. Good catch, thanks! This was C&P from a couple other places, so I'll check & fix them too.

Reply:
Oh, well, I deleted the places that this was C&P from 🤷